api/v1alpha1/servicedefaults_types_test.go

```diff
@@ -0,0 +1,455 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	capi "github.com/hashicorp/consul/api"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestRun_ToConsul(t *testing.T) {
+	cases := []ToConsulCase{
+		{
+			&ServiceDefaults{
```

Comment:
These look great but I think we can still make them a bit easier to read by only setting each attribute we actually want to test. I've started on making the code changes here since that's the easiest way to explain what i'm thinking: https://github.com/hashicorp/consul-k8s/compare/crdPOC...test-example-changes-crdPOC.

e.g.
```go
		"meshgateway.mode:remote": {
			&ServiceDefaults{
				Spec: ServiceDefaultsSpec{
					MeshGateway: MeshGatewayConfig{Mode: "remote"},
				},
			},
			&capi.ServiceConfigEntry{
				Kind:        capi.ServiceDefaults,
				MeshGateway: capi.MeshGatewayConfig{Mode: capi.MeshGatewayModeRemote},
			},
		},
```


What do you think?

Reply:
💯 improves focus of each test!! I was trying to get away with fewer shotgun style tests, but the above style makes for better readability

Reply:
Also, thanks for creating the `test-example-changes-crdPOC` branch for the clarification. It really helped. You're far too kind.